src/BloomExe/Edit/JpegWarningDialog.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using L10NSharp;
+
+namespace Bloom.Edit
+{
+	public partial class JpegWarningDialog : Form
+	{
+		public JpegWarningDialog()
+		{
+			InitializeComponent();
+			_warningText.Text = LocalizationManager.GetString("EditTab.JpegWarningDialog.WarningText",
+				@"The file you’ve chosen is a “jpeg” file. JPEGs are perfect for photographs and color artwork. However, JPEGs are a big problem for black and white line art. Problems include:
+• Fuzziness and grey dots.
+• Large file sizes, making the book hard to share.
+• If there are many large JPEGs, Bloom may not have enough memory to make PDFs.
+
+Note: Because JPEG is “lossy”, converting a JPEG to PNG, TIFF, or bmp actually makes things even worse. If this is black and white line art, you want to get an original scan in one of those formats.
+
+Please select from one of the following, then click “OK”:").Replace("\r\n","\n").Replace("\n", Environment.NewLine);
+		}
+
+		private void _okButton_Click(object sender, EventArgs e)
+		{
+			DialogResult = _cancelRadioButton.Checked ? DialogResult.Cancel : DialogResult.OK;
+			Close();
+		}
+
+		/// <summary>
+		/// Tells whether the supplied image looks suspiciously like a file that 
+		/// would be better off as a PNG. Not particularly smart, but fast.
+		/// </summary>
+		public static Boolean ShouldWarnAboutJpeg(Image image)
+		{
+			var bmp = image as Bitmap;
+			if (bmp == null)
+				return false;
+			return HasLotsOfWhiteSpace(bmp) && (IsGreyScale(bmp) || !HasLotsOfColor(bmp));
+		}
+		private static bool IsGreyScale(Bitmap bmp)
+		{
+			var sampleLinePercentages = new[] { 20, 50, 70 };
+			return
+				sampleLinePercentages.Any(sampleLine => GetIsGrey(bmp, (int)(bmp.Height * (sampleLine / 100.0))));
+		}
+		private static bool HasLotsOfColor(Bitmap bmp)
+		{
+			//what's a good threshold? 
+			//Higher than one would think, as we want to detect pictures that don't have color just because they were
+			//mistakenly made into a jpeg.
+			//A  drawing with just 4 colors had 62 once ran through jpeg.
+			// Just turning the bloom placeholder flower into jpeg gives us 10 "colors" (different shades of grey)
+			const int threshold = 100; 
+			var sampleLinePercentages = new[] { 20, 50, 70 };
+			return
+				sampleLinePercentages.Any(sampleLine => GetNumberOfColors(bmp, (int)(bmp.Height * (sampleLine/100.0))) > threshold);
+		}
+		private static bool HasLotsOfWhiteSpace(Bitmap bmp)
+		{
+			const double threshold = .5; // we'll warn if any of the samples we take of the image show > 50% white
+			var sampleLinePercentages = new[] { 20, 50, 70 };
+			return
+				sampleLinePercentages.Any(sampleLine => GetPercentWhiteOfLine(bmp, (int)(bmp.Height * (sampleLine / 100.0))) > threshold);
```

Comment:
I find myself nervous here, wondering whether on a very small image this expression could ever evaluate to a value equal to bmp.Height, which would be out of range. I don't think it can, since (int) rounds down...thought it just worth a mention.